### PR TITLE
[17.0][OU-ADD] product: migrate to 17.0

### PR DESCRIPTION
**Pull request by daiduongnguyen-odoo**

### CODE REVIEW

**Review by hbrunn — APPROVED** [review]

File: openupgrade_scripts/scripts/product/17.0.1.2/upgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+product      / product.pricelist        / message_ids (one2many)        : NEW relation: mail.message
+# NOTHING TO DO
+
+product      / product.pricelist.item   / active (boolean)              : DEL
+# NOTHING TO DO
+
+product      / product.product          / activity_user_id (many2one)   : not related anymore
+product      / product.product          / activity_user_id (many2one)   : now a function
+product      / product.product          / message_main_attachment_id (many2one): DEL relation: ir.attachment
+product      / product.product          / product_document_ids (one2many): NEW relation: product.document
```
Comment (remi-filament) [inline openupgrade_scripts/scripts/product/17.0.1.2/upgrade_analysis_work.txt:40]: Should not the attachment from product be also created in product.document model ?
I guess we should do something similar to https://github.com/odoo/odoo/blob/24c2b7bf95e494157fd4d93bbf159a04ce9f1ee6/addons/product/models/ir_attachment.py#L14-L24

Comment (daiduongnguyen-odoo) [reply openupgrade_scripts/scripts/product/17.0.1.2/upgrade_analysis_work.txt:40]: oh seem make sense

**Comment by hbrunn** [comment]
superseded by https://github.com/OCA/OpenUpgrade/pull/4598

**Pull request closed without merging**